e2e/go.mod

```diff
@@ -14,14 +14,21 @@ require (
 	k8s.io/utils v0.0.0-20240102154912-e7106e64919e
 	sigs.k8s.io/cluster-api v1.7.3
 	sigs.k8s.io/cluster-api-provider-aws/v2 v2.5.0
+	sigs.k8s.io/cluster-api-provider-azure v1.15.2
```

Comment:
Should we use https://github.com/kubernetes-sigs/cluster-api-provider-azure/releases/tag/v1.16.0 now that it available upstream and has the `DisableExtensionOperations` available? So we can remove the replace below too! :)